rasa/nlu/model.py

```diff
@@ -161,13 +161,13 @@ def _build_pipeline(
         self, cfg: RasaNLUModelConfig, component_builder: ComponentBuilder
     ) -> List[Component]:
         """Transform the passed names of the pipeline components into classes."""
-
         pipeline = []
 
         # Transform the passed names of the pipeline components into classes
-        for i in range(len(cfg.pipeline)):
-            component_cfg = cfg.for_component(i)
+        for index, pipeline_component in enumerate(cfg.pipeline):
+            component_cfg = cfg.for_component(index)
             component = component_builder.create_component(component_cfg, cfg)
+            components.validate_component_keys(component, cfg.pipeline[index])
```

Comment:
```suggestion
            components.validate_component_keys(component, pipeline_component)
```

Reply:
ah, i was wondering what the use of the second element was